### ISSUE TITLE
Bind instance of ProfitTrailer to specific IP

### ISSUE THREAD

**Issue by NOP-Commander** [taniman/profit-trailer-enhancements#274]
Dear developers,

to overcome -especially on Binance- temp bans for too many API Requests i would like to request the feature to bind an instance of ProfitTrailer to a specific IP. If i have a VPS with multiple IPs I could easily avoid API-Calls per time/per IP ban issues when I run multiple instances of ProfitTrailer on the same VPS (what I actually do on my linux VPS - but not more than 1 instance on Binance anymore).

Thanks in advance for any kind of response which comes into my favor. (;

sincerely yours,
NOP-Commander

**Comment by networkjackbb** [comment]
Yes please.

**Comment by NOP-Commander** [comment]
Just for the record: The implementation of that feature would be a "safer harbor" approach for a feature I requested over here: https://github.com/mehtadone/PTFeeder/issues/316
I know the idea of running many instances in parallel might be kinda insane but technical possible and legit for "research". (;

p.s.: I forgot to say thanks for your fine piece of software.

**Comment by clownfish44** [comment]
Multiple copy's of PT runs fine on one VPS even in Japan as of 2.0.2. Therefore this is not needed unless your running like 4+ bots.

**Comment by NOP-Commander** [comment]
Hi @clownfish44,

thanks for your feedback. Do you run 3 bots behind a single ip against a single exchange? If not, you compared apples with pears. If yes, which exchange do you target? I hope it is Binance.

If API-calls (or whatever) would be limited per IP this would potentially affect every single fellow customers of yours running more than 1 bot against a single exchange. It's theory but not plain impossible.

This request for sure implies a super-duper-low priority but is a possible and legit solution, probably even the sustainable one, for a non-existing problem (: which could come up in the future having an impact on (a part) your satisfied customers which have the possibility to setup multiple IPs on their installation to overcome a possible "per-IP restriction" in whatever way.

You could be a step ahead. Your choice. (;

yours sincerely,
NOP-Commander

**Comment by networkjackbb** [comment]
The basic facts are that PT is a daemon-style piece of software that presents a front-end to the outside world along with initiating connections to external IPs. Based on that alone, it is **standard** practice for such software to allow the administrator to bind it to a specific IP configured on the system it is running on. It allows the administrator to make certain choices in how that software runs on their network/behind their firewall, etc.
--
Be a good netizen and add this feature. I imagine that it is not that hard since your software is written in Java. In just 5 seconds I found several articles on StackOverflow talking about how to do just such a thing.

**Comment by clownfish44** [comment]
As of 2.0.4 you can only run 1 bot on 1 IP unless you throttle your bots so this may again be useful!

**Comment by LiamAird** [comment]
Possible with: http://wiki.profittrailer.com/doku.php?id=application.properties#serveraddress
